activerecord/test/cases/associations/eager_test.rb

```diff
@@ -1340,6 +1339,10 @@ def test_deep_including_through_habtm
     # has-many :through
     david = Author.where(id: "1").eager_load(:readonly_comments).first!
     assert david.readonly_comments.first.readonly?
+
+    # belongs_to
+    post = Post.where(id: "1").eager_load(:author).first!
```

Comment:
While I can't remember this problem well now, I think this behavior is not related to `readonly` option of `belongs_to`. At least `joins` was such a behavior and I unified them. I guess it is limited as readonly because the aim of eager loading is not writing but just reading.